js/bigone.js

```diff
@@ -335,40 +336,40 @@ module.exports = class bigone extends Exchange {
         //         ...
         //     ]
         //
-        const defaultSettle = this.safeString (this.options, 'defaultSettle', 'USDT')
         for (let i = 0; i < contracts.length; i++) {
             // Assumes contract is settled to defaultSettle it it's present in symbol
             // Else it assumes USD as the base and settle in base currency
             const market = contracts[i];
-            const id = this.safeString (market, 'symbol', '');
-            const useDefaultSettle = id.indexOf (defaultSettle) >= 0;
-            const quoteId = useDefaultSettle ? defaultSettle : 'USD';
-            const baseId = id.slice (0, id.indexOf (quoteId));
-            const settleId = usdt ? 'USDT' : baseId;
+            const id = this.safeString (market, 'symbol');
+            const lastChar = id[id.length - 1];
+            const difference = (lastChar === 'T') ? 4 : 3;
+            const index = id.length - difference;
+            const quoteId = id.slice (index);
+            const settleId = quoteId;
+            const baseId = id.slice (0, index);
             const base = this.safeCurrencyCode (baseId);
             const quote = this.safeCurrencyCode (quoteId);
             const settle = this.safeCurrencyCode (settleId);
             const entry = {
                 'id': id,
-                'uuid': undefined,
                 'symbol': base + '/' + quote + ':' + settle,
                 'base': base,
                 'quote': quote,
                 'settle': settle,
                 'baseId': baseId,
                 'quoteId': quoteId,
                 'settleId': settleId,
-                'type': 'spot',
+                'type': 'swap',
                 'spot': false,
                 'margin': false,
                 'swap': true,
                 'future': false,
                 'option': false,
                 'active': true,
                 'contract': true,
-                'linear': usdt,
-                'inverse': !usdt,
-                'contractSize': undefined,
+                'linear': true,
```

Comment:
@frosty00 I'm not sure if this works, as in the case of BTC/USD it settles in BTC being inverse (BTC/USD:BTC) while in the case of BTC/USDT it's linear and settles in USDT (BTC/USDT:USDT)

Reply:
Should I change it back to how it was before? Or maybe assume if the quote ends in T it's linear if not it's inverse?